### PR TITLE
fix: Dynamic Group Booking link for organization

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- Fix broken dynamic group booking links for organisations - Regression from https://github.com/calcom/cal.com/pull/12721
- Added Checkly test under organization as well as e2e tests for the same

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Tests (Unit/Integration/E2E or any other test)

## How should this be tested?

- Simply create 2 users in an organization(org1)
- Got to org1.cal.local:3000/user1+user2
- You should be able to book through that.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #12825** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12825?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12825?utm_source=stack-comment).

File: apps/web/pages/org/[orgSlug]/[user]/[type].tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const getServerSideProps = async (ctx: GetServerSidePropsContext) => {
       },
     });
   }
-  const params = { user: teamOrUserSlug, type };
+  const params = { user: teamOrUserSlugOrDynamicGroup, type };
```
Comment (hariombalhara) [inline apps/web/pages/org/[orgSlug]/[user]/[type].tsx:50]: Don't send slugified value.

File: apps/web/playwright/lib/testUtils.ts [hunk 2]
```diff
@@ -134,12 +134,16 @@ export async function bookFirstEvent(page: Page) {
   await bookEventOnThisPage(page);
 }
 
-export const bookTimeSlot = async (page: Page, opts?: { name?: string; email?: string }) => {
+export const bookTimeSlot = async (page: Page, opts?: { name?: string; email?: string; title?: string }) => {
   // --- fill form
   await page.fill('[name="name"]', opts?.name ?? testName);
   await page.fill('[name="email"]', opts?.email ?? testEmail);
+  if (opts?.title) {
```
Comment (hariombalhara) [inline apps/web/playwright/lib/testUtils.ts:141]: Support filling dynamic group booking title.

File: packages/features/ee/organizations/lib/orgDomains.ts [hunk 3]
```diff
@@ -140,7 +140,7 @@ export function whereClauseForOrgWithSlugOrRequestedSlug(slug: string) {
   } satisfies Prisma.TeamWhereInput;
 }
 
-export function userOrgQuery(hostname: string, fallback?: string | string[]) {
-  const { currentOrgDomain, isValidOrgDomain } = getOrgDomainConfigFromHostname({ hostname, fallback });
+export function userOrgQuery(req: IncomingMessage | undefined, fallback?: string | string[]) {
+  const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(req, fallback);
```
Comment (hariombalhara) [inline packages/features/ee/organizations/lib/orgDomains.ts:144]: Support running tests with custom org slug. `orgDomainConfig` has the support

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 4]
```diff
@@ -965,11 +961,12 @@ async function handler(
     throw new HttpError({ statusCode: 400, message: error.message });
   }
 
+  console.log("loading users");
```
Comment (exception) [inline packages/features/bookings/lib/handleNewBooking.ts:964]:
```suggestion
```

**Review by exception — DISMISSED** [review]

**Review by zomars — APPROVED** [review]
LGTM

**Pull request merged**